nix/ipfs-migrator.nix

```diff
@@ -0,0 +1,24 @@
+{ buildGoModule, lib }:
+
+buildGoModule rec {
+  pname = "ipfs-migrator";
+  version = "1.7.1";
+  rev = "v${version}";
+
+  src = (import ./sources.nix).ipfs-migrator;
+
+  vendorSha256 = null;
+
+  doCheck = false;
+
+  subPackages = [ "." ];
+
+  meta = with lib; {
+    description = "Migrations for the filesystem repository of ipfs clients";
+    homepage = "https://ipfs.io/";
+    license = licenses.mit;
+    platforms = platforms.unix;
+    maintainers = with maintainers; [ elitak ];
+  };
+}
+
```

Comment:
Since we were DMing about how to get rid of that `rec` (since it's considered a bit of a smell).

`rec` lets you make self-referencing sets. It's needed here so that `rev` can use `version` directly. We can tweak this along these lines:

```nix
let
  version = "1.7.1";
  pkgs = import ./sources.nix;
in
  buildGoModule { # <--- removed rec
    pname = "ipfs-migrator";
    inherit version; # i.e. version = version
    rev = "v${version}";

    src = pkgs.ipfs-migrator;
    vendorSha256 = null;
    doCheck = false;
    subPackages = [ "." ];

    meta = { # <--- got rid of `with`
      description = "Migrations for the filesystem repository of ipfs clients";
      homepage = "https://ipfs.io/";
      license = licenses.mit;
      platforms = platforms.unix;
      maintainers =  [ lib.maintainers.elitak ]; # <--- got rid of `with`
    };
}
```

I'm assuming that `lib` here exposes `lib.maintainers.elitak`. The fact that I can't tell by reading the code is why _I personally_ am not a fan of `with`, though it's very common, and different people will have different opinions.